src/check/resource/std/collection.py

```diff
@@ -1,6 +1,14 @@
 from typing import TypeVar, Generic, Union
 
-class set(Generic[T]):
+class collection(Generic[T]):
+    def __init__(self): pass
+    def __iter__(self) -> collection_iter[T]: pass
+
+class collection_iter(Generic[T]):
+    def __init__(self): pass
+    def __next__(self) -> T: pass
+
+class set(Generic[T], collection[T]):
```

Comment:
Looks like the type checker cannot deal with generics split over two parents.
At least like that.